src/qibolab/pulses/sequence.py

```diff
@@ -92,27 +94,21 @@ def coupler_pulses(self, *couplers):
         return new_pc
 
     @property
-    def finish(self) -> int:
-        """The time when the last pulse of the sequence finishes."""
-        t: int = 0
-        for pulse in self:
-            if pulse.finish > t:
-                t = pulse.finish
-        return t
-
-    @property
-    def start(self) -> int:
-        """The start time of the first pulse of the sequence."""
-        t = self.finish
+    def pulses_per_channel(self):
+        """Return a dictionary with the sequence per channel."""
+        sequences = defaultdict(self.__class__)
         for pulse in self:
-            if pulse.start < t:
-                t = pulse.start
-        return t
+            sequences[pulse.channel].append(pulse)
+        return sequences
 
     @property
     def duration(self) -> int:
-        """Duration of the sequence calculated as its finish - start times."""
-        return self.finish - self.start
+        """The time when the last pulse of the sequence finishes."""
+        channel_pulses = self.pulses_per_channel
+        if len(channel_pulses) == 1:
+            pulses = next(iter(channel_pulses.values()))
+            return sum(pulse.duration for pulse in pulses)
```

Comment:
This method is currently using recursion and this special case is the termination condition (otherwise it will go to infinite recursion). This should also be something that can be simplified by #792, particularly if `PulseSequence` is converted to a `dict` of `list`s (one for every logical channel), instead of a flattened list that is now.